core/ledger/kvledger/txmgmt/statedb/commontests/test_common.go

```diff
@@ -1014,40 +1013,58 @@ func TestFullScanIterator(
 		return sampleData
 	}
 
-	// add the sample data for four namespaces to the db
+	// add the sample data for five namespaces to the db
+	allNamesapces := stringset{"", "ns1", "ns2", "ns3", "ns4"}
 	batch := statedb.NewUpdateBatch()
-	for _, kv := range generateSampleData("", "ns1", "ns2", "ns3", "ns4") {
+	for _, kv := range generateSampleData(allNamesapces...) {
 		batch.PutValAndMetadata(kv.Namespace, kv.Key, kv.Value, kv.Metadata, kv.Version)
 	}
-	db.ApplyUpdates(batch, version.NewHeight(5, 5))
+	sourceDB.ApplyUpdates(batch, version.NewHeight(5, 5))
 
-	// verifyFullScanIterator verifies the output of the FullScanIterator with skipping zero or more namespaces
-	verifyFullScanIterator := func(skipNamespaces stringset) {
-		fullScanItr, valFormat, err := db.GetFullScanIterator(
+	// verifyExportImport uses FullScanIterator (with skipping zero or more namespaces)
+	// for exporting the data to import into another ledger instance and verifies the
+	// correctness of the imported data
+	verifyExportImport := func(destDBName string, skipNamespaces stringset) {
+		fullScanItr, valFormat, err := sourceDB.GetFullScanIterator(
 			func(ns string) bool {
 				return skipNamespaces.contains(ns)
 			},
 		)
 		require.NoError(t, err)
 		require.Equal(t, valueFormat, valFormat)
+
+		destinationDB, err := dbProvider.GetDBHandle(destDBName, nil)
+		assert.NoError(t, err)
+
+		err = destinationDB.ImportState(fullScanItr, valFormat)
+		require.NoError(t, err)
+
+		for _, nsNotToBePresent := range skipNamespaces {
+			iter, err := destinationDB.GetStateRangeScanIterator(nsNotToBePresent, "", "")
+			require.NoError(t, err)
+			res, err := iter.Next()
+			require.NoError(t, err)
+			require.Nil(t, res)
+		}
+
+		expectedNamespacesInDestinationDB := allNamesapces.minus(skipNamespaces)
 		results := []*statedb.VersionedKV{}
-		for {
-			compositeKV, serializedVersionedValue, err := fullScanItr.Next()
+		for _, nsToBePresent := range expectedNamespacesInDestinationDB {
+			iter, err := destinationDB.GetStateRangeScanIterator(nsToBePresent, "", "")
 			require.NoError(t, err)
-			if compositeKV == nil {
-				break
+			for {
+				res, err := iter.Next()
+				require.NoError(t, err)
+				if res == nil {
+					break
+				}
+				versionedKV := res.(*statedb.VersionedKV)
+				results = append(results, versionedKV)
 			}
-			versionedVal, err := dbValueDeserializer(serializedVersionedValue)
-			require.NoError(t, err)
-			results = append(results, &statedb.VersionedKV{
-				CompositeKey:   *compositeKV,
-				VersionedValue: *versionedVal,
-			})
 		}
 
-		expectedNamespacesInResults := stringset{"", "ns1", "ns2", "ns3", "ns4"}.minus(skipNamespaces)
 		expectedResults := []*statedb.VersionedKV{}
-		for _, ns := range expectedNamespacesInResults {
+		for _, ns := range expectedNamespacesInDestinationDB {
 			expectedResults = append(expectedResults, generateSampleData(ns)...)
 		}
 		require.Equal(t, expectedResults, results)
```

Comment:
can be replaced wtih `require.Equal(t, generateSampleData(expectedNamespacesInDestinationDB...), results)`

Reply:
yes, this makes it shorter